core/play/src/main/scala/play/api/mvc/Action.scala

```diff
@@ -432,6 +432,12 @@ trait ActionBuilder[+R[_], B] extends ActionFunction[Request, R] {
       def apply(request: Request[A]) =
         try {
           invokeBlock(request, block)
+            .andThen { _ =>
+              // This andThen block is used to keep a reference to the request until invokeBlock() is completed.
+              // If the request contains TemporaryFiles, they will be deleted when they are garbage collected.
+              // By keeping a reference to the request, it prevents the TemporaryFiles from becoming GC targets.
+              request
+            }(self.executionContext)
```

Comment:
We can use trampoline here to avoid thread switching, because the code that get's executed is a total no-op.
```suggestion
            }(Execution.trampoline)
```